src/libs/ReportUtils.ts

```diff
@@ -7754,9 +7753,9 @@ function hasOnlyHeldExpenses(iouReportID: string): boolean {
 /**
  * Checks if thread replies should be displayed
  */
-function shouldDisplayThreadReplies(reportAction: OnyxInputOrEntry<ReportAction>, reportID: string): boolean {
+function shouldDisplayThreadReplies(reportAction: OnyxInputOrEntry<ReportAction>, isThreadFirstChatParam: boolean): boolean {
```

Comment:
Can you please explain a bit more about why we still need to use the isThreadFirstChat function, and therefore need the Param suffix here? You've said that the reportID associated with actions that have threads now point to the report where they are actually created. That's a frontend only change right? Since an action's reportID should always point to the report it's attached to.

I see the function is still used in getOriginalReportID, but I don't really understand the logic there? Why do we only use isThreadFirstChat if the currentReportAction is empty?

Reply:
> You've said that the reportID associated with actions that have threads now point to the report where they are actually created. That's a frontend only change right?

Yes

> Since an action's reportID should always point to the report it's attached to.

Report action object doesn't have a reportID.
<img width="315" alt="image" src="https://github.com/user-attachments/assets/750a21df-fcc9-40d2-993a-349625ab83fb">


So, `getOriginalReportID` want to see whether the passed `reportAction` params belongs to the `reportID` params.

```
const reportActions = allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`];
const currentReportAction = reportActions?.[reportAction?.reportActionID ?? '-1'] ?? null;
```

If the report action belongs to the `reportID`, then it will just return the `reportID`. Otherwise, it will use `isThreadFirstChat`. I didn't remove `isThreadFirstChat` from `getOriginalReportID` because of this [reason](https://github.com/Expensify/App/pull/51721#issuecomment-2446473423).

Reply:
I think we can just extract the contents of isThreadFirstChat directly into getOriginalReportID, and since that's exactly the same thing it won't have any effect. Does that sound good?

Reply:
Oh, that's a good idea. Updated